File: M241.Server/Data/Models/RoomData.cs
﻿using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace M241.Server.Data.Models
{
    public class RoomData
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("humidity")]
        public float Humidity { get; set; }

        [JsonPropertyName("temperature")]
        public float Temperature { get; set; }

        [JsonPropertyName("pressure")]
        public float Pressure { get; set; }

        [JsonPropertyName("gas")]
        public float Gas { get; set; }

        [JsonPropertyName("timeStamp")]
        public DateTime TimeStampLocal => TimeStamp.ToLocalTime();

        [JsonPropertyName("timeStampUTC")]
        public DateTime TimeStamp { get; set; }

        [JsonPropertyName("room")]
        public Room? Room { get; set; }

        [JsonPropertyName("roomId")]
        public int RoomId { get; set; }

        [JsonPropertyName("isBurning")]
        [NotMapped]
        public bool IsBurning { get; set; }
    }
}

File: M241.Frontend/models/RoomData.ts
import type Room from "./Room";

class RoomData {
  id: number;
  humidity: number;
  temperature: number;
  pressure: number;
  gas: number;
  isBurning: boolean;
  timeStamp: string;
  room: Room | null;
  roomId: number;

  constructor(
    id: number,
    humidity: number,
    temperature: number,
    pressure: number,
    gas: number,
    timeStamp: string,
    roomId: number,
    room: object,
    isBurning: boolean
  ) {
    this.id = id;
    this.humidity = humidity;
    this.temperature = temperature;
    this.pressure = pressure;
    this.gas = gas;
    this.timeStamp = timeStamp;
    this.roomId = roomId;
    this.room = room;
    this.isBurning = isBurning;
  }
}

export default RoomData;
